### PR TITLE
CSHARP-4939: Add VectorSearchScore builder for $vectorSearch stage

### PR DESCRIPTION
**Pull request by BorisDog**
Also added typed field expression overloads  for Score* helpers

### CODE REVIEW

File: src/MongoDB.Driver/ProjectionDefinitionBuilder.cs [hunk 1]
```diff
@@ -184,12 +201,46 @@ public static ProjectionDefinition<TDocument> Meta<TDocument>(this ProjectionDef
         /// </returns>
         public static ProjectionDefinition<TDocument> MetaSearchScore<TDocument>(
             this ProjectionDefinition<TDocument> projection,
-            string field)
+            FieldDefinition<TDocument> field)
```
Comment (sanych-sun) [inline src/MongoDB.Driver/ProjectionDefinitionBuilder.cs:204]: Is this intentional breaking change (here and some more signature changes below)?

Comment (BorisDog) [reply src/MongoDB.Driver/ProjectionDefinitionBuilder.cs:204]: Sorry, should have put a comment.
It's not breaking as `FieldDefinition<TDocument>` has implicit conversion from string.

**Review by adelinowona — APPROVED** [review]
LGTM!

**Review by sanych-sun — APPROVED** [review]
LGTM

**Pull request merged**